### PR TITLE
0.6.1/Auth Client

### PR DESCRIPTION
**Pull request by k7Xm9pQr**
Base on #5 

Please briefly answer these questions:

* what problem are you trying to solve? (or if there's no problem, what's the motivation for this change?)
    Add auth client
* what changes does this pull request make?
    Add auth client
* are there any non-obvious implications of these changes? (does it break compatibility with previous versions, etc)
    No

### CODE REVIEW

File: client/src/main/java/cloud/xline/client/AuthClient.java [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+        CommandResponse resp = this.curpClient.propose(cmd, true);
+        return resp.getAuthStatusResponse();
+    }
+
+    public AuthenticateResponse authenticate(AuthenticateRequest req) throws InvalidProtocolBufferException, CommandExecutionException {
+        RequestWithToken reqToken = RequestWithToken.newBuilder().setToken(this.token).setAuthenticateRequest(req).build();
+        Command cmd = Command.newBuilder().setRequest(reqToken).build();
+        CommandResponse resp = this.curpClient.propose(cmd, false);
+        return resp.getAuthenticateResponse();
+    }
```
Comment (bsbds) [inline client/src/main/java/cloud/xline/client/AuthClient.java:44]: The `authenticate` operation does need to set token.

File: client/src/main/java/cloud/xline/client/AuthClient.java [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+        RequestWithToken reqToken = RequestWithToken.newBuilder().setToken(this.token).setAuthUserAddRequest(req).build();
+        Command cmd = Command.newBuilder().setRequest(reqToken).build();
+        CommandResponse resp = this.curpClient.propose(cmd, false);
+        return resp.getAuthUserAddResponse();
+    }
+
+    public AuthUserGetResponse userGet(AuthUserGetRequest req) throws InvalidProtocolBufferException, CommandExecutionException {
+        RequestWithToken reqToken = RequestWithToken.newBuilder().setToken(this.token).setAuthUserGetRequest(req).build();
+        Command cmd = Command.newBuilder().setRequest(reqToken).build();
+        CommandResponse resp = this.curpClient.propose(cmd, false);
```
Comment (bsbds) [inline client/src/main/java/cloud/xline/client/AuthClient.java:56]: `userGet` and `roleGet` could use fast path.

File: client/src/main/java/cloud/xline/client/AuthClient.java [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+    }
+
+    public AuthenticateResponse authenticate(AuthenticateRequest req) throws InvalidProtocolBufferException, CommandExecutionException {
+        RequestWithToken reqToken = RequestWithToken.newBuilder().setToken(this.token).setAuthenticateRequest(req).build();
+        Command cmd = Command.newBuilder().setRequest(reqToken).build();
+        CommandResponse resp = this.curpClient.propose(cmd, false);
+        return resp.getAuthenticateResponse();
+    }
+
+    public AuthUserAddResponse userAdd(AuthUserAddRequest req) throws InvalidProtocolBufferException, CommandExecutionException {
```
Comment (bsbds) [inline client/src/main/java/cloud/xline/client/AuthClient.java:46]: Please hash the password in `userAdd` and `userChangePassword`

**Comment by k7Xm9pQr** [comment]
Close because unused, see #7

**Pull request closed without merging**